subsys/bluetooth/host/hci_core.c

```diff
@@ -5921,13 +5920,32 @@ int bt_le_adv_start_internal(const struct bt_le_adv_param *param,
 		if (err) {
 			return err;
 		}
+
+		if (IS_ENABLED(CONFIG_BT_PERIPHERAL) &&
+		    param->options & BT_LE_ADV_OPT_CONNECTABLE) {
+			conn = bt_conn_add_le(param->id, BT_ADDR_LE_NONE);
+			if (!conn) {
+				return -ENOMEM;
+			}
+
+			bt_conn_set_state(conn, BT_CONN_CONNECT_ADV);
+		}
 	}
 
 	err = set_advertise_enable(true);
 	if (err) {
+		BT_ERR("Failed to start advertiser");
+		if (IS_ENABLED(CONFIG_BT_PERIPHERAL) && conn) {
+			bt_conn_set_state(conn, BT_CONN_DISCONNECTED);
+			bt_conn_unref(conn);
+		}
 		return err;
 	}
 
+	if (IS_ENABLED(CONFIG_BT_PERIPHERAL) && conn) {
```

Comment:
Same here - I suspect the `IS_ENABLED()` can be dropped

Reply:
If you compare this with the reference handling of `bt_conn_create_le` (or slave) then this is the ref that would otherwise have been given to the application, and the application would have the responsebility of calling bt_conn_unref. This one balances the ref made by bt_conn_add_le.

The ref that is maintained by the stack is in the bt_conn_set_state(conn, BT_CONN_CONNECT_ADV);
This one will released when bt_conn_set_state(conn, BT_CONN_DISCONNECTED); transitions from a non-connected state.

To clarify
```
conn = bt_conn_add_le(param->id, BT_ADDR_LE_NONE); // Ref 1

if (!conn) {
	return -ENOMEM;
}

bt_conn_set_state(conn, BT_CONN_CONNECT_ADV); // Ref 2

err = set_advertise_enable(true);
if (err) {
	bt_conn_set_state(conn, BT_CONN_DISCONNECTED); // Ref 1
	bt_conn_unref(conn); // Ref 0
	return err; // Conn object freed on return
}

bt_conn_unref(conn); // Ref 1
return 0; // Conn object has valid reference, freed by bt_le_adv_stop, or disconnected callback


```

I can add this comment:
"Since we don't give the application a reference to manage in this case, we need to release this reference here."

Reply:
Added comment for cases where we create the conn, but don't give it to the application to manage.